packages/upswyng-web/src/components/ReportIssue.tsx

```diff
@@ -0,0 +1,194 @@
+import { useHistory, useParams } from "react-router-dom";
+
+import Button from "@material-ui/core/Button";
+import Checkbox from "@material-ui/core/Checkbox";
+import { Container } from "../App.styles";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormGroup from "@material-ui/core/FormGroup";
+import Input from "@material-ui/core/Input";
+import LoadingSpinner from "./LoadingSpinner";
+import PageBanner from "./PageBanner";
+import React from "react";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
+import axios from "axios";
+import createStyles from "@material-ui/core/styles/createStyles";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import { useLastLocation } from "react-router-last-location";
+import useResource from "./useResource";
+
+interface State {
+  [index: string]: {
+    text: string;
+    checked: boolean;
+  };
+}
+
+let serverUri = process.env.REACT_APP_SERVER_URI || "http://localhost:3000";
+if (serverUri.charAt(serverUri.length - 1) === "/") {
+  serverUri = serverUri.slice(0, -1);
+}
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    marginTop: {
+      marginTop: "2rem",
+    },
+  })
+);
+
+const ReportIssue = () => {
+  const { resourceId } = useParams();
+  const resource = useResource(resourceId || "");
+  const history = useHistory();
+  const lastLocation = useLastLocation();
+  const classes = useStyles();
+  const [other, setOther] = React.useState("");
+  const [comments, setComments] = React.useState("");
+  const [submissionError, setError] = React.useState(false);
+  const [formSubmitted, setFormSubmitted] = React.useState(false);
+  const [issues, updateIssues] = React.useState({
+    address: {
+      text: "The address is wrong.",
+      checked: false,
+    },
+    phone: {
+      text: "The phone number is wrong.",
+      checked: false,
+    },
+    website: {
+      text: "The website link is wrong.",
+      checked: false,
+    },
+    schedule: {
+      text: "The schedule is wrong.",
+      checked: false,
+    },
+    directions: {
+      text: "The directions are wrong.",
+      checked: false,
+    },
+    other: {
+      text: "Other",
+      checked: false,
+    },
+  } as State);
```

Comment:
Hmm, all these slices of state are smelly. Maybe we could use [Formik](https://jaredpalmer.com/formik/) here to handle our state and submissions. Mr. CTO, @rhinodavid, thoughts?